### PR TITLE
Actually submit task graph logs and node IDs to the server.

### PR DESCRIPTION
**Pull request by thetorpedodog**
This comprises most (all?) of the rest of the client-side work for
logging the activities of client-driven task graphs.

- We build the task graph structure and submit it to the server.
- For each graph execution step, we include the node metadata.
- When the graph finishes, submit that we are complete.

### CODE REVIEW

File: tiledb/cloud/dag/dag.py [hunk 1]
```diff
@@ -946,10 +1027,10 @@ def maybe_replace(self, arg) -> Optional[visitor.Replacement]:
 
 
 def _topo_sort(
-    lst: Sequence[models.TaskGraphNodeMetadata],
-) -> Sequence[models.TaskGraphNodeMetadata]:
+    lst: Sequence[rest_api.TaskGraphNodeMetadata],
+) -> Sequence[rest_api.TaskGraphNodeMetadata]:
     """Topographically sorts the list of node metadatas."""
```
Comment (johnkerl) [inline tiledb/cloud/dag/dag.py:1032]: Should be `Topologically`

**Review by johnkerl — APPROVED** [review]
LGTM, and there are unit-test cases.

🚢

File: tiledb/cloud/dag/dag.py (hunk 1 above)
Comment (thetorpedodog) [reply tiledb/cloud/dag/dag.py:1032]: maybe i want to sort the list of metadatas by rolling it down a hill. you don’t know my life

File: tiledb/cloud/dag/dag.py [hunk 2]
```diff
@@ -368,17 +425,33 @@ def execute_update_callbacks(self):
...
+            return
+        try:
+            api_st = _API_STATUSES[self.status]
+        except KeyError as ke:
+            raise ValueError(
+                f"Task graph ended in invalid state {self.status!r}"
+            ) from ke
+        try:
+            client.client.task_graph_logs_api.update_task_graph_log(
+                id=str(self.server_graph_uuid),
```
Comment (thetorpedodog) [inline tiledb/cloud/dag/dag.py:448]: After running a test against staging, I had to make one more change here, to add the `str` conversion. The generated client doesn’t like trying to serialize `uuid.UUID`s.

**Comment by thetorpedodog** [comment]
The one erroring test is a transient. Merging.

**Pull request merged**